modules/tinymce/src/core/test/ts/browser/delete/CefDeleteTest.ts

```diff
@@ -140,4 +140,33 @@ describe('browser.tinymce.core.delete.CefDeleteTest', () => {
       })
     );
   });
+
+  context('cef is at the start/end of the content and covered with selection', () => {
+    it('TINY-7795: should delete selected content when cef block at the start', () => {
+      const editor = hook.editor();
+      editor.setContent('<p contenteditable="false">CEF</p><p>abc</p>');
+      // actual content: <p data-mce-caret="before"></p><p contenteditable="false">CEF</p><p>abc</p>
+      TinySelections.setSelection(editor, [], 0, [ 2, 0 ], 2);
+      TinyContentActions.keystroke(editor, Keys.backspace());
+      TinyAssertions.assertCursor(editor, [ 0, 0 ], 0);
+      TinyAssertions.assertContent(editor, '<p>c</p>');
+    });
+
+    it('TINY-7795: should delete selected content when cef block at the end', () => {
+      const editor = hook.editor();
+      editor.setContent('<p>abc</p><p contenteditable="false">CEF</p>');
+      TinySelections.setSelection(editor, [ 0, 0 ], 1, [], 2);
+      TinyContentActions.keystroke(editor, Keys.backspace());
+      TinyAssertions.assertCursor(editor, [ 0, 0 ], 1);
+      TinyAssertions.assertContent(editor, '<p>a</p>');
+    });
+
+    it('TINY-7795: should delete selected content when cef block at the start and at the end', () => {
+      const editor = hook.editor();
+      editor.setContent('<p contenteditable="false">CEF</p><p>abc</p><p contenteditable="false">CEF</p>');
+      editor.execCommand('SelectAll');
+      TinyContentActions.keystroke(editor, Keys.backspace());
+      TinyAssertions.assertContent(editor, '');
+    });
+  });
```

Comment:
As mentioned elsewhere this definitely isn't part of the scope for TINY-7795. I'm guessing this is something you found? If so it should be logged as a separate Jira and done in that instead of grouping it in with this change.

Reply:
![TINY-7795](https://user-images.githubusercontent.com/44773772/169831417-b8d7915d-4076-40e7-9609-21f356b54e96.png)

Reply:
I've created follow up TINY-8729

Reply:
@vpyshnenko sorry for the confusion here, I didn't mean it had to be deleted from this PR, only that it wasn't about navigation and as such should have been a separate Jira with an associated changelog entry. I have no issues with the 2 being done in this single PR.

As for the text, given I was in the call that was created for... the intention/meaning was to make sure the selection was in such a state that deletion would still work, not to create a custom deletion override. I haven't looked into it enough if we actually need to the override though but I'm assuming it was given your investigation.

Reply:
As discussed on slack, we're going to leave it as 2 separate PRs since Viktor has already reverted this change.